src/three/plugins/images/sources/GeoJSONImageSource.js

```diff
@@ -0,0 +1,423 @@
+
+export class GeoJSONImageSource extends TiledImageSource {
+
+	constructor( {
+
+		geojson = null,
+		url = null, // URL or GeoJson object can be provided
+		tileDimension = 256,
+		bounds = [ - 180, - 90, 180, 90 ],
+		levels = 18,
+		projection = null, // 'EPSG:3857' or 'EPSG:4326' or null to auto- detect
+		pointRadius = 6,
+		strokeStyle = 'red',
+		fillStyle = 'rgba( 255,0,0,0.25 )'
+
+	} = {} ) {
+
+		super( );
+		this.geojson = geojson;
+		this.url = url;
+		this.tileDimension = tileDimension;
+		this.bounds = bounds;
+		this.levels = levels;
+		this.projection = projection;
+		this.pointRadius = pointRadius;
+		this.strokeStyle = strokeStyle;
+		this.fillStyle = fillStyle;
+
+	}
+
+	async init( ) {
+
+		// configure tiling to content bounds and levels
+		this.tiling.setProjection( new ProjectionScheme( this._detectProjection( ) ) );
+		this.tiling.setContentBounds( ...this.bounds );
+		for ( let i = 0; i < this.levels; i ++ ) {
+
+			const tilesX = 2 ** i;
+			const tilesY = 2 ** i;
+			this.tiling.setLevel( i, {
+				tilePixelWidth: this.tileDimension,
+				tilePixelHeight: this.tileDimension,
+				tileCountX: tilesX,
+				tileCountY: tilesY,
+			} );
+
+		}
+
+		// If a URL was provided and no geojson object yet, fetch it now (use fetchData so overlay can inject headers)
+		if ( ! this.geojson && this.url ) {
+
+			try {
+
+				const res = await this.fetchData( this.url );
+				// fetchData should return a Response-like object
+				// allow both Response and plain object (in case overlay fetch returns parsed JSON)
+				if ( res && typeof res.json === 'function' ) {
+
+					this.geojson = await res.json();
+
+				} else if ( res ) {
+
+					this.geojson = res;
+
+				}
+
+			} catch ( e ) {
+
+				console.warn( 'GeoJSONImageSource: failed to fetch geojson from url', this.url, e );
+				// continue, but without geojson no tiles will be produced
+
+			}
+
+		}
+
+		return Promise.resolve( );
+
+	}
+
+	// main fetch per tile - > returns THREE.Texture
+	async fetchItem( tokens, signal ) {
+
+		const [ x, y, level ] = tokens;
+		const boundsNormalized = this.tiling.getTileBounds(
+			x,
+			y,
+			level,
+			true,
+			false,
+		); // [ minx,miny,maxx,maxy ] normalized
+
+		// compute tile extent in projection units ( degrees for 4326, meters for 3857 )
+		const tileInfo = this._tileExtentFromNormalized( boundsNormalized );
+
+		// create canvas
+		const canvas = document.createElement( 'canvas' );
+		canvas.width = this.tileDimension;
+		canvas.height = this.tileDimension;
+		const ctx = canvas.getContext( '2d' );
+
+		// clear
+		ctx.clearRect( 0, 0, canvas.width, canvas.height );
+
+		// find features that ( quick bbox ) intersect tile
+		const features =
+			this.geojson && this.geojson.features ? this.geojson.features : [ ];
+
+		for ( let i = 0; i < features.length; i ++ ) {
+
+			const f = features[ i ];
+			if ( ! f || ! f.geometry ) continue;
+			if ( ! this._featureIntersectsTile( f, tileInfo ) ) continue;
```

Comment:
I wonder if we can cache the bounding boxes for each feature instead of recalculating it ever draw.